doc/sphinx/proof-engine/tactics.rst

```diff
@@ -898,206 +1111,154 @@ Applying theorems
+            Goal forall x y : nat, x = y -> y = x.
 
-         On the subgoal :g:`forall x y : nat, x = y -> y = x` the
-         tactic :n:`intros until 1` is equivalent to :n:`intros x y H`,
-         as :g:`x = y -> y = x` is the first non-dependent product.
+         .. coqtop:: all
 
-         On the subgoal :g:`forall x y z : nat, x = y -> y = x` the
-         tactic :n:`intros until 1` is equivalent to :n:`intros x y z`
-         as the product on :g:`z` can be rewritten as a non-dependent
-         product: :g:`forall x y : nat, nat -> x = y -> y = x`.
+            intros until y.
 
-      .. exn:: No such hypothesis in current goal.
+         Or:
 
-         This happens when :token:`natural` is 0 or is greater than the number of
-         non-dependent products of the goal.
+         .. coqtop:: reset out
 
-   .. tacv:: intro {? @ident__1 } after @ident__2
-             intro {? @ident__1 } before @ident__2
-             intro {? @ident__1 } at top
-             intro {? @ident__1 } at bottom
+            Goal forall x y : nat, x = y -> y = x.
 
-      These tactics apply :n:`intro {? @ident__1}` and move the freshly
-      introduced hypothesis respectively after the hypothesis :n:`@ident__2`,
-      before the hypothesis :n:`@ident__2`, at the top of the local context,
-      or at the bottom of the local context. All hypotheses on which the new
-      hypothesis depends are moved too so as to respect the order of
-      dependencies between hypotheses. It is equivalent to :n:`intro {? @ident__1 }`
-      followed by the appropriate call to :tacn:`move`,
-      :tacn:`move … before …`, :tacn:`move … at top`,
-      or :tacn:`move … at bottom`.
+         .. coqtop:: all
 
-      .. note::
+            intros until 1.
 
-         :n:`intro at bottom` is a synonym for :n:`intro` with no argument.
+      .. exn:: No quantified hypothesis named @ident in current goal even after head-reduction.
 
-      .. exn:: No such hypothesis: @ident.
-         :undocumented:
+         The :n:`@ident` in the ``until`` clause doesn't appear as a :term:`dependent premise`.
 
-.. tacn:: intros {* @intropattern }
-   :name: intros …
+      .. exn:: No @natural-th non dependent hypothesis in current goal even after head-reduction.
 
-   Introduces one or more variables or hypotheses from the goal by matching the
-   intro patterns.  See the description in :ref:`intropatterns`.
+         There are fewer than :n:`@natural` premises in the goal.
 
 .. tacn:: eintros {* @intropattern }
-   :name: eintros
 
-   Works just like :tacn:`intros …` except that it creates existential variables
-   for any unresolved variables rather than failing.
+   Works just like :tacn:`intros` except that it creates existential variables
+   for any unresolved variables rather than failing.  Typically this happens when
+   using a ``%`` intropattern (see :n:`@simple_intropattern`).
 
-.. tacn:: clear @ident
+.. tacn:: clear {| {? {? - } {+ @ident } } | @natural }
```

Comment:
Since I've noticed something else to change above, let me point out that this `clear` line, ideally should have been kept unchanged in this PR. But no big deal if it isn't.